### DISCUSSION TITLE
Where can I find all predefinitions of Display Type and Location in  OrchardCore source code, please?

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#15778] [General]
Where can I find all predefinitions of Display Type and Location in  OrchardCore source code, please?

**Comment by Piedone** [comment]
If you mean where can you find a list of these in the code, then you can't: display types are not predefined but are strings that are passed to Orchard's display management (see [here](https://github.com/OrchardCMS/OrchardCore/blob/9b7468b9ddd869bb9a11bdbc1fecfd30cf2a1355/src/OrchardCore/OrchardCore.DisplayManagement/IDisplayManager.cs#L9) and for content items [here](https://github.com/OrchardCMS/OrchardCore/blob/9b7468b9ddd869bb9a11bdbc1fecfd30cf2a1355/src/OrchardCore/OrchardCore.ContentManagement.Display/IContentItemDisplayManager.cs#L13)). By convention, the most important ones are "Detail", "Summary", and "SummaryAdmin", but anything else is possible (and used).

**Comment by effapp** [comment]
Hi @Piedone ,

I was wondering how many display types and Locations(example:  "Content: 5" or "ActionsMenu:3") can I use?

**Reply by hishamco** [reply]
Please elaborate

**Reply by Piedone** [reply]
Yeah, sorry but I don't understand either.

**Reply by effapp** [reply]
I mean this:

![image](https://github.com/OrchardCMS/OrchardCore/assets/51013754/67ece717-d0ad-452f-bff1-b94563efdca2)

**Reply by Piedone** [reply]
I told about the common display types above. Apart from those, you can use any string. On how these matter, see [here](https://docs.orchardcore.net/en/latest/docs/reference/modules/Themes/#display-types) and [here](https://github.com/search?q=repo%3ALombiq%2FOrchard-Training-Demo-Module+placement&type=code).

For locations, see [here](https://docs.orchardcore.net/en/latest/docs/reference/modules/Themes/#content-zones-differentiators).
